backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/ServerApiErrorHandlingWrapper.java

```diff
@@ -0,0 +1,260 @@
+  }
+
+  public HotspotApi.HotspotsPullResult pullHotspots(String projectKey, String branchName, Set<SonarLanguage> enabledLanguages, @Nullable Long changedSince,
+    SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.hotspot().pullHotspots(projectKey, branchName, enabledLanguages, changedSince, cancelMonitor));
+  }
+
+  public ServerHotspotDetails showHotspot(String hotspotKey, SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.hotspot().show(hotspotKey, cancelMonitor));
+  }
+
+  public Collection<ServerHotspot> getAllServerHotspots(String projectKey, String branchName, SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.hotspot().getAll(projectKey, branchName, cancelMonitor));
+  }
+
+  public void anticipatedTransitions(String sonarProjectKey, List<LocalOnlyIssue> localOnlyIssues, SonarLintCancelMonitor cancelMonitor) {
+    callWithErrorHandleAndRethrow(() -> serverApi.issue().anticipatedTransitions(sonarProjectKey, localOnlyIssues, cancelMonitor));
+  }
+
+  public void changeIssueStatus(String issueKey, Transition reviewStatus, SonarLintCancelMonitor cancelMonitor) {
+    callWithErrorHandleAndRethrow(() -> serverApi.issue().changeStatus(issueKey, reviewStatus, cancelMonitor));
+  }
+
+  public void changeHotspotStatus(String hotspotKey, HotspotReviewStatus newStatus, SonarLintCancelMonitor cancelMonitor) {
+    callWithErrorHandleAndRethrow(() -> serverApi.hotspot().changeStatus(hotspotKey, newStatus, cancelMonitor));
+  }
+
+  public boolean supportHotspotsPull(Version version) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.hotspot().supportHotspotsPull(version));
+  }
+
+  public Collection<ServerHotspot> getHotspotsForFile(String projectKey, Path serverFilePath, String branchName, SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.hotspot().getFromFile(projectKey, serverFilePath, branchName, cancelMonitor));
+  }
+
+  public Optional<ServerHotspotDetails> tryFetchHotspot(String hotspotKey, SonarLintCancelMonitor cancelMonitor) {
+    return serverApi.hotspot().fetch(hotspotKey, cancelMonitor);
+  }
+
+  public ServerStatusInfo getSystemStatus(SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.system().getStatus(cancelMonitor));
+  }
+
+  public Optional<ServerRule> tryFetchRule(String ruleKey, SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.rules().getRule(ruleKey, cancelMonitor));
+  }
+
+  public Optional<IssueApi.ServerIssueDetails> tryFetchIssue(String issueKey, String projectKey, String branch, @Nullable String pullRequest,
+    SonarLintCancelMonitor cancelMonitor) {
+    return callWithErrorHandleAndRethrow(() -> serverApi.issue().fetchServerIssue(issueKey, projectKey, branch, pullRequest, cancelMonitor));
```

Comment:
It works in the sense that it is a transparent wrapper that will notify the client in case a wrapped call throws a corresponding exception. If the wrapped call doesn't throw - the wrapper won't do anything.